packages/block-library/src/columns/edit.js

```diff
@@ -124,15 +202,13 @@ export default withDispatch( ( dispatch, ownProps, registry ) => ( {
 		setAttributes( { columns } );
 
 		let innerBlocks = getBlocks( clientId );
-		if ( ! hasExplicitColumnWidths( innerBlocks ) ) {
```

Comment:
There's a bug here in master if you try to change the number of columns from `2` to `4` (more than 1). This function only accounted for a single addition/removal.